docs/develop/gettingstarted.rst

```diff
@@ -0,0 +1,285 @@
+   Use one of the commands below:
+
+.. code-block:: console
+
+   $ git clone https://github.com/micropython/micropython.git 
+   $ git clone https://github.com/<your-user-name>/micropython.git
+
+Depending on what you cloned, `configure the remote 
+repositories <https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes>`_ to be able to
+collaborate on the MicroPython project.
+
+Configure upstream if you cloned your fork.
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+
+It is common to configure ``upstream`` and ``origin`` on a forked repository
+to assist with sharing code changes. You can maintain your own mapping but
+it is recommended that ``origin`` maps to your fork and ``upstream`` to the main
+MicroPython repository:
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+   $ git remote set-url origin https://github.com/<your-user-name>/micropython.git
+
+After the above configuration, your setup should be similar to this:
+
+.. code-block:: console
+
+   origin	https://github.com/<your-user-name>/micropython.git (fetch)
+   origin	https://github.com/<your-user-name>/micropython.git (push)
+   upstream	https://github.com/micropython/micropython.git (fetch)
+   upstream	https://github.com/micropython/micropython.git (push)
+
+You should now have a copy of the source code. By default you are pointing
+to the master branch. To prepare for further development, it is recommended
+to work on a development branch.
+
+.. code-block:: console
+
+    $ git checkout -b dev-branch
+
+You can give it any name. You will have to compile MicroPython whenever you change 
+to a different branch.
+
+Compile and Build the Code
+--------------------------
+
+For development, you should compile MicroPython for a specific platform which corresponds
+to a port. Before building for any port, first build the MicroPython cross-compiler.
+You should also install the required dependencies for the port in order to sucessfully
+compile and build.
+
+Build the cross-compiler
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Almost all ports require building ``mpy-cross`` first to perform pre-compilation
+of scripts that will be included in the port firmware:
+
+.. code-block:: console
+
+   $ cd mpy-cross
+   $ make
+
+If ``mpy-cross`` built successfully, you should see a message similar to this:
+
+.. code-block:: console
+
+   LINK mpy-cross
+      text	   data	    bss	    dec	    hex	filename
+    279328	    776	    880	 280984	  44998	mpy-cross
+
+Build for the Unix Port
+~~~~~~~~~~~~~~~~~~~~~~~
+To build for the unix port, make sure you have a working environment for ``gcc`` and ``GNU make``.
+Ubuntu 20.04 has been used for the example below but other unixes ought to work with little
+modification:
+
+.. code-block:: console
+
+   $ gcc --version
```

Comment:
It could be confusing that this will only work after the installation listed below, more so for building of the cross-compiler. Also Python (best to recommend Python 3 I guess) needs to be installed.

Reply:
What are you suggesting then? That we move details of this dependency to before building the cross-compiler?

Reply:
Definitely before it, yes, since those are all prerequisite. As far as I know people who read this just want to get things working so it makes sense that it's a document which you start from the top end read to the bottom, following steps in the order presented.

Actually there could even a separate section which bundles the prerequisites, i.e. the ones mentioned now, and Python, and git as well because it's not really different from the other applications needed.